comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go

```diff
@@ -99,6 +106,8 @@ func newProvides(deps dependencies) provides {
 	return provides{
 		Comp:           c,
 		StatusProvider: status.NewInformationProvider(autodiscoveryStatus.GetProvider(c)),
+
+		Endpoint: api.NewAgentEndpointProvider(c.(*AutoConfig).writeConfigCheck, "/config-check", "GET"),
```

Comment:
Do we need the type casting here?

Reply:
I think we do because `newAutoConfig` returns a `autodiscovery.Component` and not a `*AutoConfig`, it does not compile otherwise

Reply:
I'm okay with it, because we're handling all of the construction within these couple of functions.